stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -1203,6 +1202,10 @@ function(add_swift_target_library_single target name)
 
   list(APPEND library_search_directories "${SWIFT_SDK_${sdk}_ARCH_${arch}_PATH}/usr/lib/swift")
 
+  # Override -arch on macOS. This is crucial for Apple Silicon, where CMake forces
+  # the *host* architecture to appear as the argument to -arch otherwise.
+  set_property(TARGET ${target} PROPERTY OSX_ARCHITECTURES "${SWIFTLIB_SINGLE_ARCHITECTURE}")
```

Comment:
@edymtt I'm not sure if this is the appropriate place to put it, but we aren't passing `target` down to `_add_target_variant_c_compile_link_flags` AFAICT so it's a bit tough to move there.